### PR TITLE
Deprecate ReportingExtension.baseDir

### LINKED ISSUE #20803: Deprecate ReportingExtension.baseDir
It has been replaced by `getBaseDirectory()` in Gradle 4.4 but was never deprecated.

---
cc: @gradle/execution

### PR DESCRIPTION
**Pull request by ov7a**
Fixes #20803

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by ov7a** [comment]
@bot-gradle test this

**Review by asodja — APPROVED** [review]
LGTM

File: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc [hunk 1]
```diff
@@ -165,6 +165,14 @@ For example, `jvmArgs "some", "arg"` can be replaced with `jvmArgs("some", "arg"
 ==== DependencyInsightReportTask.getDependencySpec
 The method was deprecated because it was not intended for public use in build scripts.
 
+[[reporting-base-dir]]
+==== ReportingExtension.baseDir
+`ReportingExtension.getBaseDir()`, ``ReportingExtension.setBaseDir(File)`, and `ReportingExtension.setBaseDir(Object)` were deprecated.
+They should be replaced with `ReportingExtension.getBaseDirectory()` property.
+
+[[reporting-base-dir]]
```
Comment (ljacomet) [inline platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:173]: ❌ Looks like a left over section anchor

**Review by ljacomet — CHANGES_REQUESTED** [review]

**Review by big-guy — APPROVED** [review]

File: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc (hunk 1 above)
Comment (ov7a) [reply platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc]: Thanks!

**Pull request merged**